CMakeLists.txt

```diff
@@ -117,10 +117,26 @@ else()
   message(WARNING "Java Development component or JNI not found, JNI targets will not work")
 endif()
 
-# Use the new MSVC preprocessor to improve standard conformance.
-if (CMAKE_CXX_COMPILER_ID MATCHES "MSVC") 
+if (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+  # Use the new MSVC preprocessor to improve standard conformance.
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /Zc:preprocessor")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:preprocessor")
+
+  # Starting from MSVC 14.34, the following warning was introduced:
```

Comment:
I don't think we need quite so much context in the comment. I would recommend removing the example and instead linking to the change that added the warning.

Reply:
Simplified comment. :)